### PR TITLE
Fix 'use 5.x'

### PR DESCRIPTION
**Pull request by atoomic**

### CODE REVIEW

File: cpan/ExtUtils-Constant/t/Constant.t [hunk 1]
```diff
@@ -409,7 +409,11 @@ EOT
   push @files, $pm;
   open FH, ">$pm" or die "open >$pm: $!\n";
   print FH "package $package;\n";
-  print FH "use $];\n";
+  if ( "$]" < 7 ) {
+    print FH "use $];\n";
+  } else {
+    print FH "use v".int($]).";\n";
```
Comment (toddr) [inline cpan/ExtUtils-Constant/t/Constant.t:415]: use $^V

File: t/comp/use.t [hunk 2]
```diff
@@ -69,6 +69,50 @@ sub ok($;$) {
...
+    'no v5',
+    'use 5.34',
+    'use v5.34',
+    'use 5.33',
+    'use v5.33',
+);
+
+my @good = (
+    'use v7',
+    'no v7',
```
Comment (toddr) [inline t/comp/use.t:97]: This should be in the bad camp. Nobody uses it. what does it mean anyway?

**Comment by atoomic** [comment]
This is going to be its own PR in blead at 7.1.0

File: t/comp/use.t [hunk 3]
```diff
@@ -69,6 +69,50 @@ sub ok($;$) {
...
+    'use v6.0.1',
+    'use 6.001',
+    'no 6',
+    'no v6',
+    'use 66',
+    'no 66',
+    # use v7.*; # index('v7.')
+    'use 7',
+    'use 7.0',
+    'use 7.42', # index('7')
```
Comment (toddr) [inline t/comp/use.t:85]: use v7.22.0

File: pp_ctl.c [hunk 4]
```diff
@@ -3753,23 +3753,62 @@ static OP *
...
+	        case '4':
+	        case '5':
+	            break;
+	        case '6':
+	            DIE(aTHX_ "'use %s' is not supported by Perl 7", str_version );
+	            break;
+	        case '7':
+	            /* use 7* is not supported */
+	            if (ptr == str_version || strlen(ptr) != 1)
+                    DIE(aTHX_ "use v7; is the only supported syntax for Perl 7." );
```
Comment (toddr) [inline pp_ctl.c:3785]: Add to the pod file which documents errors.

**Comment by jkeenan** [comment]
@atoomic, has this pull request been superseded by our work on Objective 2a?

Thank you very much.
Jim Keenan

**Comment by atoomic** [comment]
closing this pull request as we have already provided work in this area to the alpha branch

**Pull request closed without merging**